bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java

```diff
@@ -250,26 +282,13 @@ private void checkForUpdatedVersion(File idf_json_file)
 			JSONObject jsonObj = (JSONObject) parser.parse(reader);
 			String idfVersionId = (String) jsonObj.get(IDF_VERSIONS_ID);
 			JSONObject list = (JSONObject) jsonObj.get(IDF_INSTALLED_LIST_KEY);
-			if (list == null)
-			{
-				return;
-			}
-			// selected esp-idf version information
-			JSONObject selectedIDFInfo = (JSONObject) list.get(idfVersionId);
-			String idfPath = (String) selectedIDFInfo.get(IDF_PATH);
-			IDFEnvironmentVariables idfEnvMgr = new IDFEnvironmentVariables();
-			if (idfEnvMgr.getEnvValue(IDFEnvironmentVariables.IDF_PATH).equals(idfPath))
-				return;
-			newIdfPath = idfPath;
-			Preferences prefs = getPreferences();
-			prefs.putBoolean(IS_INSTALLER_CONFIG_SET, false);
-			try
-			{
-				prefs.flush();
-			}
-			catch (BackingStoreException e)
+			if (list != null)
 			{
-				Logger.log(e);
+				// selected esp-idf version information
+				JSONObject selectedIDFInfo = (JSONObject) list.get(idfVersionId);
+				String idfPath = (String) selectedIDFInfo.get(IDF_PATH);
+				IDFEnvironmentVariables idfEnvMgr = new IDFEnvironmentVariables();
+				return !idfEnvMgr.getEnvValue(IDFEnvironmentVariables.IDF_PATH).equals(idfPath);
```

Comment:
idfEnvMgr.getEnvValue(IDFEnvironmentVariables.IDF_PATH) will return null when IDE is not configured with the IDF_PATH(IDE is not launched and configured not even once). That leads to NPE

Reply:
In case of multiple back slashes in the path, this might fail.
IDE would have configured IDF_PATH to `C:\esp\esp-idf-5.0` and windows installer would have set to `C:\\esp\\esp-idf-5.0` as per equals() both will be different

Reply:
good catch. Probably we could use Paths.get().normalize() here

Reply:
> good catch. Probably we could use Paths.get().normalize() here

I dont think using normalize is a good idea simply comparison like using `Paths.get(path).toAbsolutePath().compareTo(pathToCompare) == 0`
should be enough for us considering that we null check first.